.github/workflows/email.yml

```diff
@@ -51,7 +51,11 @@ jobs:
               echo "$diff" 
               echo 'DIFF<<EOF' >> $GITHUB_ENV
               echo "$diff" >> $GITHUB_ENV
-              echo 'EOF' >> $GITHUB_ENV                   
+              echo 'EOF' >> $GITHUB_ENV  
+
+              head_sha=$(git show HEAD --format=%T | head -1)
```

Comment:
This uses the hash of the tree pointed to by HEAD, instead of the hash of HEAD itself, which is unexpected to me. Does it not work properly with just HEAD sha?

Reply:
(i.e., `$(git show HEAD --format=%H | head -1)` or just `$(git rev-parse HEAD)`)

Reply:
Both have the same result.
git show HEAD --format=%H | head -1
0c77f390bb1baf8dafa9a7a02cad00151938bc5a
(base) bhavani@Bhavanis-MacBook-Pro-2 cron % git rev-parse HEAD
0c77f390bb1baf8dafa9a7a02cad00151938bc5a

Reply:
Yes, that's what I'm saying, either of those would give you HEAD sha, whereas the command being used to calculate "head_sha" currently actually calculates SHA of the tree of head.

Reply:
Both gives the same sha
 
 git show HEAD --format=%H | head -1
0c77f390bb1baf8dafa9a7a02cad00151938bc5a
 % git rev-parse HEAD
0c77f390bb1baf8dafa9a7a02cad00151938bc5a

Reply:
`git show HEAD --format=%T | head -1` (my suggestion) is not the same thing as `git show HEAD --format=%T | head -1` (your code). The format specifier is different. I'm well aware that the two examples I gave you of how to get the HEAD sha both give the HEAD sha.

Reply:
git show HEAD --format=%H | head -1 points to the latest commit sha in my branch vs we want the tip of the main chapel repo. 
397da5e

Reply:
As that command depends on what your `HEAD` is and I know nothing about the context you're running that command in it doesn't tell me anything in particular.

The difference between the commands is not what commit they point to -- the command you are using does not give the SHA of *any* commit, it gets the SHA of the *tree* pointed to by HEAD. Are you familiar with git trees and is it what you mean to be retrieving the hash of?

Reply:
Yes, I am familiar with git trees and I am trying to get the SHA of the tree pointed by HEAD to construct the compare url (compare the changes between the tip of the PR branch and the main branch(SHA of the tree pointed by HEAD)).

https://git-scm.com/docs/gitglossary#gitglossary-aiddefcommit-ishacommit-ishalsocommittish

Reply:
Understood. I was wondering because I assumed from the variable name `head_sha` that it was intended to be that commit hash. In that case, is there a reason for preferring the tree hash over commit hash? Does using one vs the other give different results for what is shown on the github page you generate a link to?

Reply:
(also, I think that git glossary link might be broken, because it looks like it has anchors for a specific part of the glossary but it just lead me to the top of the page when I clicked on it)

Reply:
Without a rebase, the compare url constructed from the tree hash is also incorrect(Getting the chapel-main branch tip/tree hash is not pointing to the latest when not remerged) I am going to go ahead and close this PR.
Brad also mentioned that having the diff is good enough and most of them are using PR link to get the changes.

I also tried to construct the compare url by using the branch names for ex.
https://github.com/chapel-lang/chapel/compare/main...bhavanijayakumaran:chapel:compare_url , this works for an unmerged PR but does not work on a PR that is already merged to main.
Email workflow runs on a merged PR so we cannot use branch names instead of the commit.